src/UpdateEmail.js

```diff
@@ -0,0 +1,96 @@
+import React from "react";
+import {withRouter, Link} from 'react-router-dom'
+import {getCookie} from './cookies';
+import Header from './Header/Header'
+
+
+class UpdateEmail extends React.Component {
+    constructor(props) {
+      super(props);
+      this.state = {
+        userId: getCookie("userId"),
+        email: "",
+        error:"",
+      };
+      
+      this.handleInputChange = this.handleInputChange.bind(this);
+      this.handleSubmit = this.handleSubmit.bind(this);
+      
+    }
+    
+    handleInputChange(event) {
+        const target = event.target;
+        const value = target.value;
+        const name = target.name;
+
+        this.setState({
+          [name]: value,
+        });
+
+    }
+        
+    handleSubmit = (event) => {
+        event.preventDefault();
+        
+        if (this.state.email != ""){
+
+            const id = this.state.userId;
+            const myForm = new FormData (document.getElementById("newEmailForm"));
+            myForm.append("id", id);
+
+            const myRequest = new Request("http://127.0.0.1:5000/updateEmail", {
```

Comment:
can you check if the old email is equal to the new email before calling the back-end?

Reply:
it is now checked at backend. Front end doesnt have the info of old email i think.